### PR TITLE
(#77) Simplify BoundedByteBuffer

### PR DESCRIPTION
**Pull request by vzurauskas**
This is for https://github.com/yegor256/cactoos-http/issues/77.

I refactored the `BoundedByteBuffer` to make it simpler.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#85](https://github.com/yegor256/cactoos-http/issues/85) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#85](https://github.com/yegor256/cactoos-http/issues/85) is assigned to @ilyakharlamov[/z](https://www.0crat.com/u/ilyakharlamov), here is [why](https://www.0crat.com/footprint/C63314D6Z/57b474a4-0f42-4f3d-a285-7465e75ca2b2); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/main/java/org/cactoos/http/io/BoundedByteBuffer.java [hunk 1]
```diff
@@ -76,35 +85,39 @@ public void offer(final byte add) {
...
-                    break;
-                }
-                result = bytes[idn] == this.internal[idn];
+        if (this.size() != bytes.length) {
+            return false;
+        }
+        int index = this.start;
+        for (final byte current : bytes) {
+            if ((int) current != (int) this.internal[index]) {
+                return false;
```
Comment (ilyakharlamov) [inline src/main/java/org/cactoos/http/io/BoundedByteBuffer.java]: This line `return false` is never tested, add a unit test for this case.

File: src/main/java/org/cactoos/http/io/BoundedByteBuffer.java [hunk 2]
```diff
@@ -76,35 +85,39 @@ public void offer(final byte add) {
...
+
+    /**
+     * Size of this buffer.
+     * @return This buffer's size.
+     */
+    private int size() {
+        // @checkstyle AvoidInlineConditionalsCheck (3 lines)
+        return this.full
+            ? this.internal.length
+            : this.empty() ? 0 : this.end;
```
[the remark this answers is not in the document]
Comment (vzurauskas) [reply src/main/java/org/cactoos/http/io/BoundedByteBuffer.java]: @ilyakharlamov Changed to `if`s. Not sure if it's easier to read this way, but ok.

File: src/test/java/org/cactoos/http/io/BoundedByteBufferTest.java [hunk 3]
```diff
@@ -69,9 +68,23 @@ public void worksWithSmallerArrayFailsComparisonWithLarger() {
...
             buffer.equalTo(new byte[] {1, 2, 0}),
             new IsEqual<>(false)
         );
     }
+
+    @Test
+    public void equalsWhenShiftedOdd() {
+        final int limit = 4;
+        final BoundedByteBuffer buffer = new BoundedByteBuffer(limit);
+        final byte[] bytes = {1, 2, 1, 2};
```
[the remark this answers is not in the document]
Comment (vzurauskas) [reply src/test/java/org/cactoos/http/io/BoundedByteBufferTest.java]: Ok.

File: src/test/java/org/cactoos/http/io/BoundedByteBufferTest.java [hunk 4]
```diff
@@ -69,9 +68,23 @@ public void worksWithSmallerArrayFailsComparisonWithLarger() {
...
+        buffer.offer((byte) 11);
+        for (final byte curr : bytes) {
```
[the remark this answers is not in the document]
Comment (vzurauskas) [reply src/test/java/org/cactoos/http/io/BoundedByteBufferTest.java]: @ilyakharlamov Changed.

File: src/main/java/org/cactoos/http/io/BoundedByteBuffer.java (hunk 1 above)
Comment (vzurauskas) [reply src/main/java/org/cactoos/http/io/BoundedByteBuffer.java]: @ilyakharlamov Added.

**Comment by ilyakharlamov** [comment]
@llorllale good to merge in my opinion

File: src/main/java/org/cactoos/http/io/BoundedByteBuffer.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (ilyakharlamov) [reply src/main/java/org/cactoos/http/io/BoundedByteBuffer.java]: This is discussable, I see no problem with it, and can understand what's going on in double ternary. But I've seen some developers, even some good developers who just don't understand ternaries at all, even a single ternary is a problem for them, they understand the ternary conditions the other way around.

[the remark this answers is not in the document]
Comment (vzurauskas) [reply src/main/java/org/cactoos/http/io/BoundedByteBuffer.java]: I get that. I personally think ternaries, when used right, make code more concise and declarative, and just like programmers who find it difficult to understand object oriented code should just learn it, so the programmers who find it difficult to understand ternaries should learn them. However, the benefits of ternaries (unlike object orientation) are subjective, so I think code reviewer's opinion combined with project's static analysis settings trumps my opinion.

[the remark this answers is not in the document]
Comment (ilyakharlamov) [reply src/main/java/org/cactoos/http/io/BoundedByteBuffer.java]: agreed, there are pros and cons in it
I do not mind a single ternary but double was probably too much here.
[SO](https://stackoverflow.com/questions/160218/to-ternary-or-not-to-ternary)

**Comment by vzurauskas** [comment]
@llorllale ping

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
Code review was too long (21 days), architects (@llorllale) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by 0crat** [comment]
The job [#85](https://github.com/yegor256/cactoos-http/issues/85) is now out of scope

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @ilyakharlamov[/z](https://www.0crat.com/u/ilyakharlamov), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by sereshqua** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @ilyakharlamov[/z](https://www.0crat.com/u/ilyakharlamov)

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

**Pull request merged**